gcloud-java-contrib/gcloud-java-nio/src/test/java/com/google/cloud/storage/contrib/nio/CloudStorageLateInitializationTest.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright 2016 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.storage.contrib.nio;
+
+import com.google.cloud.storage.StorageOptions;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+import java.net.URI;
+
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+/**
+ * Unit tests for {@link CloudStorageFileSystemProvider} late initialization.
+ */
+@RunWith(JUnit4.class)
+public class CloudStorageLateInitializationTest {
+
+  @Rule public final ExpectedException thrown = ExpectedException.none();
+
+  private StorageOptions mockOptions;
+
+  @Before
+  public void before() {
+    mockOptions = mock(StorageOptions.class);
+    when(mockOptions.service()).thenReturn(StorageOptions.defaultInstance().service());
```

Comment:
As you might have noticed tests are failing. Reason is that this code is issuing requests against the actual service in a unit test. There's no need to use `StorageOptions.defaultInstance().service()`, another mock would do just fine.